### DISCUSSION TITLE
NOT_HTML error in lighthouse

### DISCUSSION THREAD

**Discussion by sidgoel85** [GoogleChrome/lighthouse#12097] [Q&A]
Hi, 
While running lighthouse as part of the build process I'm getting below error 

Error_Code - NOT_HTML
Error_Message - the page provided is not html (served as mime type )

Any help will be greatly appreciated.

**Comment by patrickhulce** [comment]
What help are you seeking @sidgoel85? The error message explains that the page wasn't served with a MIME type that indicates it was HTML. Whatever server serves the page you're running Lighthouse against should return HTML pages with a `text/html` content type header.

**Reply by sidgoel85** [reply]
So, I'm running lighthouse as part of the Jenkins pipeline and instantiating a server using npm script and hitting the localhost URL of the page I'm evaluating which is then failing because of the above error. However, when I try to browse the page locally it works without the issues. So, if the MIME type header is missing the page should fail to load locally as well.

**Reply by patrickhulce** [reply]
> So, if the MIME type header is missing the page should fail to load locally as well.

This isn't true in practice. This error in Lighthouse is to prevent unauditable resources from being analyzed even if Chrome and other browsers are able to do error recovery, MIME sniffing, and manage to display something. There are many resources which will load in a browser that Lighthouse cannot run against.

If this page is truly valid HTML, the server you're using should set the content type header indicating that. If it's a public npm package, I'm sure they'd be open to such a fix :)

**Reply by patrickhulce** [reply]
@sidgoel85 are you using NUXT by chance? You might be able to help out in #12125 if you can provide that extra info.
